### PR TITLE
Fix CSI sanity check - Create Volume

### PR DESCRIPTION
**Pull request by c3y1huang**
Part of and relate to https://github.com/longhorn/longhorn-manager/pull/825#issuecomment-783294412.

- Add volume name autocorrection.
- Correct misuse of the error codes.

Fix https://github.com/longhorn/longhorn/issues/2270

### CODE REVIEW

File: manager/volume.go [hunk 1]
```diff
@@ -258,7 +259,7 @@ func (m *VolumeManager) Create(name string, spec *types.VolumeSpec) (v *longhorn
 
 	v = &longhorn.Volume{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: name,
+			Name: strings.ToLower(name),
```
Comment (c3y1huang) [inline manager/volume.go:262]: At existing VolumeCreate, Longhorn will validate if the volume name is compliant with DNS (RFC 1123). So I assume there shouldn't be any existing volume named out of the boundary.

**Review by shuo-wu — CHANGES_REQUESTED** [review]
1. It's better to move `strings.ToLower` into function `AutoCorrectName`
2. Explicitly call `AutoCorrectName` before invoking the datastore function `VolumeCreate`. Function `checkVolume` should be responsible for validating the name only.

**Comment by joshimoo** [comment]
I agree with Shuo's comment, autocorrect should be moved at the beginning of the `func (m *VolumeManager) Create` function then have check only validate the name.

**Comment by c3y1huang** [comment]
> 1. It's better to move `strings.ToLower` into function `AutoCorrectName`
> 2. Explicitly call `AutoCorrectName` before invoking the datastore function `VolumeCreate`. Function `checkVolume` should be responsible for validating the name only.

> I agree with Shuo's comment, autocorrect should be moved at the beginning of the `func (m *VolumeManager) Create` function then have check only validate the name. Other than that LGTM, feel free to merge after addressing the feedback.

@shuo-wu , @joshimoo I've addressed the comments. PTAL.

File: util/util.go [hunk 2]
```diff
@@ -347,6 +347,20 @@ func SplitStringToMap(str, separator string) map[string]struct{} {
 	return ret
 }
 
+// AutoCorrectName converts name to lowercase, and
+// correct overlength name by replaces the name suffix
+// with 8 char from its checksum to ensure uniqueness.
+func AutoCorrectName(name string, maxLength int) string {
```
Comment (shuo-wu) [inline util/util.go:353]: Add a log if the name is different after the auto correction

**Review by shuo-wu — CHANGES_REQUESTED** [review]
Need to rename the 3rd commit. Others LGTM

File: util/util.go (hunk 2 above)
Comment (c3y1huang) [reply util/util.go:352]: Addressed.

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**